packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -462,6 +463,16 @@ class _HelperErrorState extends State<_HelperError> with SingleTickerProviderSta
   }
 }
 
+/// Defines the behaviour of the floating label
+enum FloatLabelBehavior {
```

Comment:
Please rename this enum and the corresponding variables (and etc) FloatingLabelBehavior. That way the name matches the way the term is used (for example on the previous line) as well as the old name, hasFloatingPlaceholder. I realize this just a fine point, but it's only easy to change once :-).